jax/_src/core.py

```diff
@@ -2803,13 +2803,6 @@ def ctx_factory():
     msg = "\n\n".join([msg, "while checking jaxpr:", jaxpr_str])
     raise JaxprTypeError(msg) from None
 
-  # Run key reuse checker after validating jaxpr:
-  if config.enable_key_reuse_checks.value:
-    # Import here to avoid circular imports
-    from jax.experimental.key_reuse._core import check_key_reuse_jaxpr  # pytype: disable=import-error
-    check_key_reuse_jaxpr(jaxpr)
```

Comment:
Removal makes sense in that this check is redundant with an eager check on jit dispatch (similarly cond, etc.). On the other hand, the redundancy shouldn't cost us because of caching, so we could maybe actually leave this in? That way someone using `check_jaxpr` without (compiling and) dispatching it still gets the optional reuse check.

Reply:
Done!